### ISSUE TITLE
Can't load new column properties after table loaded

### ISSUE THREAD

**Issue by Saturn2888** [l-lin/angular-datatables#172]
Is there a way to reload column properties after the table's been loaded?

I've got tabs setup in my project and when clicking a tab, it sends a new data object in the resolve of a promise. The issue is when clicking one of the tabs, the render function on one of the columns changes, but it's not being run to show currency instead of numbers.

I noticed you could do `$scope.dtOptions.fnPromise = ()` like you did here: https://l-lin.github.io/angular-datatables/#/dataReloadWithPromise. But I believe the intended way is `$scope.dtOptions.withFnPromise()`. This is important because if you can do that, then there must be some way to modify the column properties after the table's loaded such as `$scope.dtOptions.aoColumns = table.columns`.

In `angular-datatables.js:420`, I see `angular.extend(options, dtOptions);`. Does this have anything to do with why options aren't mutable after the fact?

**Comment by l-lin** [comment]
These helpers (`withXXX()`) are not mandatory.
Writing
```js
$scope.dtOptions = DTOptionsBuilder.newOptions()
    .withPaginationType('full_numbers')
    .withDisplayLength(2);
```
is the same as writing
```js
$scope.dtOptions = {
    paginationType: 'full_numbers',
    displayLength: 2
};
```

Can you provide a plnkr or something alike to illustrate your issue?

**Comment by Saturn2888** [comment]
I cannot easily do that no. Here's an example of what I'm doing:

```
setSelectedTable = function() {
    $scope.dtColumns = table.columns;
    $scope.dtOptions.withFnPromise(deferredReportFactory.getPromise(tableData));
};
```

`deferredReportFactory` takes JSON data and creates a promise then pushes that promise into `dtOptions`. I am able to swap this data just fine so this part is working as I'd like.

When I swap out `dtColumns` though, the column definitions are the same as the initial directive load, they don't get altered. So if one of the columns was setup to render numbers and that same column in the next table has the values as currency, they will render as numbers, not currency.

Is it possible to change this so I'm able to setup `dtColumns` again once the table is loaded if I reset the scope variable?

I even tried doing:
```
$scope.dtOptions.withOption('aoColumns', table.columns)
```
and
```
$scope.dtOptions.withOption('columns', table.columns)
```
But these didn't work either.

**Comment by l-lin** [comment]
In the current version, you cannot do it. There were an issue (#139) about the fact that the promise renderer cannot change the options.

In the upcoming version (v0.4.0), you will be able to change them like [this](http://embed.plnkr.co/pSVW18CGFeGjgknCL5Tc/).

**Comment by Saturn2888** [comment]
Ah yeah, that's pretty cool! How can I get a hold of this version right now to test it out? Is it in a stable-enough state?

**Comment by l-lin** [comment]
It's all on the `dev` branch. There are still some [development](https://github.com/l-lin/angular-datatables/milestones/v0.4.0) left before releasing a new version.

**Comment by Saturn2888** [comment]
Using that branch as a test, it fails to load any table data from the promise and no errors. Has anything changed other than the removal of the dtOptions.reloadData() function?

**Comment by l-lin** [comment]
Take a look at the plnkr on my previous post. It's using the `dev` version. Try to imitate the code.

**Comment by Saturn2888** [comment]
I spend a long time doing a revamp of my codebase with the newly released 0.4.0  and it still doesn't work.

Essentially, I have the same exact functionality as before. I can change out the data, but the columns are exactly the same. I tried using the `vm` method of `this` on the controller, I received no change in behavior from doing `$scope` so I reverted back to `$scope`.

When I do `$scope.dtColumns = table.columns` after changing the value of `table.columns`, the value of `scope.dtColumns` is correct but the column rendering is still doing with the old rendering method.

In my particular case, all of the columns are the same name, they just render differently depending on the data I'm swapping in so in one instance they're numbers, in another they're currency. But with the way it is now, they're either rendering as numbers or currency.

I even tried doing `$scope.dtInstance.rerender()` on the instance but I get the exact same result. I'm wondering if your Plunkr and website examples work because you're not modifying the rendering method.

Removing the `rerender()`, I'm getting this to work only part of the time now. Sometimes it works and displays table data with the correct render function, sometimes it does not and shows the data then quickly hides it. The time where it hides the data and shows `No data available in table` seems to be when switching the render function, but it's only sometimes.
